symengine/polynomial.h

```diff
@@ -8,9 +8,253 @@
 #include <symengine/expression.h>
 #include <symengine/monomials.h>
 
+// Calculates bit length of number, used in UIntDict*= only
+template <typename T>
+unsigned int bit_length(T t)
+{
+    unsigned int count = 0;
+    while (t > 0) {
+        count++;
+        t = t >> 1;
+    }
+    return count;
+}
+
 namespace SymEngine
 {
 
+class UIntDict
```

Comment:
1. Yes
2. I mentioned this to a comment you made in another PR? Let me know if you can't find it.
3. I feel that we should make sure the ones we do have works perfectly before going to more classes. So, keep the unordered map implementations to the future.